src/server.c

```diff
@@ -3904,6 +3909,9 @@ void call(client *c, int flags) {
     size_t zmalloc_used = zmalloc_used_memory();
     if (zmalloc_used > server.stat_peak_memory)
         server.stat_peak_memory = zmalloc_used;
+
+    /* Do some maintenance job and cleanup */
+    afterCommand(c);
```

Comment:
I didn't found any `signalModifiedKey` in `handleClientsBlockedOnKeys`.  I think those events are triggered in commands which make the keys ready. e.g. `LPUSH` make key ready which triggers invalidation message, so `handleClientsBlockedOnKeys` needs not handle invalidation message.

Reply:
So I think there's a bug with LMOVE (modifies a key without a notification)? 
I can't verify that (currently AFK), please check

Reply:
It has notification in `lmoveHandlePush`.

Reply:
So doesn't that mean that handleClientsBlockedOnKeys can schedule a notification? I which case we want to call afterCommand in handleClientsBlockedOnKeys?

Reply:
Currently this only happens with `LIST`. So should we add it in `serveClientsBlockedOnListKey` after each client is handled?

Reply:
i'm sorry, i just now realize that `handleClientsBlockedOnKeys` doesn't have a "clients" loop, and you need a client in order to call `afterCommand`.
i think you need to call `afterCommand` for all types, not only list, since for a blocked command, that's where the command is actually executed.
for now, it would be meaningful only for lists, but still we need to call it for all of them.
i think you should look for the calls to `updateStatsOnUnblock` and pair each of them with an `afterCommand` call.